### PR TITLE
Refactor file names and module names to opensearch

### LINKED ISSUE #10: Remove tests, code changes that are not related oss
logstash-output-elasticsearch can use non oss features of ES. Since they won't be supported in logstash-output-opensearch, we need to remove this .

### PR DESCRIPTION
**Pull request by VijayanB**
### Description

- Add build status badge
- Update github workflow
- Refactor ci scripts
- Remove reference to 7.x version
- Update template manager
- Update scripts to test against opensearch
- Remove versions
- Rename module names
- Add license headers
- Rename plugin folder
- Rename from elasticsearch to opensearch

 
### Issues Resolved
#10 #3 
 
### Check List
- [X] Commits are signed per the DCO using --signoff 

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

File: lib/logstash/outputs/opensearch.rb [hunk 1]
```diff
@@ -128,7 +137,7 @@ class LogStash::Outputs::ElasticSearch < LogStash::Outputs::Base
 
   config :document_type,
     :validate => :string,
-    :deprecated => "Document types are being deprecated in Elasticsearch 6.0, and removed entirely in 7.0. You should avoid this feature"
+    :deprecated => "Document types are being deprecated in OpenSearch 6.0, and removed entirely in 7.0. You should avoid this feature"
```
Comment (jmazanec15) [inline lib/logstash/outputs/opensearch.rb:140]: Is this line still necessary?

File: lib/logstash/outputs/opensearch/http_client.rb [hunk 2]
```diff
@@ -385,12 +394,14 @@ def template_exists?(name)
...
     def template_put(name, template)
       path = "#{template_endpoint}/#{name}"
-      logger.info("Installing Elasticsearch template", name: name)
+      logger.info("Installing OpenSearch template", name: name)
       @pool.put(path, nil, LogStash::Json.dump(template))
     end
 
     def template_endpoint
-      maximum_seen_major_version < 8 ? '_template' : '_index_template'
+      # TODO: Check Version < 7.8 and use index template for >= 7.8 & OpenSearch
```
Comment (jmazanec15) [inline lib/logstash/outputs/opensearch/http_client.rb:402]: Why cant we do this now?

File: lib/logstash/outputs/opensearch/http_client/pool.rb [hunk 3]
```diff
@@ -433,15 +443,13 @@ def set_last_es_version(version, url)
     end
 
     def set_maximum_seen_major_version(major)
-      if major >= 6
-        @logger.warn("Detected a 6.x and above cluster: the `type` event field won't be used to determine the document _type", es_version: major)
-      end
+      @logger.warn("Detected a 6.x and above cluster: the `type` event field won't be used to determine the document _type", es_version: major)
```
Comment (jmazanec15) [inline lib/logstash/outputs/opensearch/http_client/pool.rb:446]: Why do we need this warning?

File: lib/logstash/plugin_mixins/opensearch/api_configs.rb [hunk 4]
```diff
@@ -1,22 +1,31 @@
...
+#  compatible open source license.
+#
+#  Modifications Copyright OpenSearch Contributors. See
+#  GitHub history for details.
+
+module LogStash; module PluginMixins; module OpenSearch
   module APIConfigs
 
-    # This module defines common options that can be reused by alternate elasticsearch output plugins such as the elasticsearch_data_streams output.
+    # This module defines common options that can be reused by alternate OpenSearch output plugins such as the OpenSearch_data_streams output.
```
Comment (jmazanec15) [inline lib/logstash/plugin_mixins/opensearch/api_configs.rb:13]: I believe data streams are xpack feature. Can this be removed?

File: lib/logstash/plugin_mixins/opensearch/api_configs.rb [hunk 5]
```diff
@@ -120,10 +129,10 @@ module APIConfigs
         # See https://hc.apache.org/httpcomponents-client-ga/httpclient/apidocs/org/apache/http/impl/conn/PoolingHttpClientConnectionManager.html#setValidateAfterInactivity(int)[these docs for more info]
         :validate_after_inactivity => { :validate => :number, :default => 10000 },
 
-        # Enable gzip compression on requests. Note that response compression is on by default for Elasticsearch v5.0 and beyond
+        # Enable gzip compression on requests. Note that response compression is on by default for OpenSearch v5.0 and beyond
```
Comment (jmazanec15) [inline lib/logstash/plugin_mixins/opensearch/api_configs.rb:132]: No OpenSearch v5.0

File: lib/logstash/plugin_mixins/opensearch/common.rb [hunk 6]
```diff
@@ -1,9 +1,18 @@
...
+#  Modifications Copyright OpenSearch Contributors. See
+#  GitHub history for details.
+
+require "logstash/outputs/opensearch/template_manager"
+
+module LogStash; module PluginMixins; module OpenSearch
   module Common
 
-    # This module defines common methods that can be reused by alternate elasticsearch output plugins such as the elasticsearch_data_streams output.
+    # This module defines common methods that can be reused by alternate elasticsearch output plugins such as the OpenSearch_data_streams output.
```
Comment (jmazanec15) [inline lib/logstash/plugin_mixins/opensearch/common.rb:15]: elasticsearch name here

File: spec/fixtures/_nodes/7x.json [hunk 7]
```diff
@@ -4,7 +4,7 @@
     "successful" : 3,
     "failed" : 0
   },
-  "cluster_name" : "elasticsearch",
+  "cluster_name" : "opensearch",
```
Comment (jmazanec15) [inline spec/fixtures/_nodes/7x.json:7]: Does this path somehow need to differentiate between Elasticsearch and OpenSearch (i.e. what happens when we get to OpenSearch 7x?

Also, I thought we were removing references to 7x in this PR?

File: spec/opensearch_spec_helper.rb [hunk 8]
```diff
@@ -1,14 +1,23 @@
...
+#  The OpenSearch Contributors require contributions made to
+#  this file be licensed under the Apache-2.0 license or a
+#  compatible open source license.
+#
+#  Modifications Copyright OpenSearch Contributors. See
+#  GitHub history for details.
+
 require_relative './spec_helper'
 
 require 'elasticsearch'
```
Comment (jmazanec15) [inline spec/opensearch_spec_helper.rb:12]: We still need elasticsearch in this file?

File: lib/logstash/outputs/opensearch.rb (hunk 1 above)
Comment (VijayanB) [reply lib/logstash/outputs/opensearch.rb:140]: Good catch. I will remove 6.0 . In this PR i am not removing any code which is being used by ES 7.10.2. I will do it next PR. What do you think?

File: lib/logstash/outputs/opensearch/http_client.rb (hunk 2 above)
Comment (VijayanB) [reply lib/logstash/outputs/opensearch/http_client.rb:402]: Sure. I created a ticket here: https://github.com/opensearch-project/logstash-output-opensearch/issues/12 for discussion. Can you comment on ticket about what do you think there?

File: lib/logstash/plugin_mixins/opensearch/api_configs.rb (hunk 5 above)
Comment (VijayanB) [reply lib/logstash/plugin_mixins/opensearch/api_configs.rb:132]: Ack

File: lib/logstash/plugin_mixins/opensearch/common.rb (hunk 6 above)
Comment (VijayanB) [reply lib/logstash/plugin_mixins/opensearch/common.rb:15]: Ack

File: spec/opensearch_spec_helper.rb (hunk 8 above)
Comment (VijayanB) [reply spec/opensearch_spec_helper.rb:12]: For integration tests, we are using elasticsearch ruby api to create,delete,update index.

File: lib/logstash/outputs/opensearch/http_client/pool.rb (hunk 3 above)
Comment (VijayanB) [reply lib/logstash/outputs/opensearch/http_client/pool.rb:446]: Kept warning. Will scan for relevancy in next PR.

File: lib/logstash/plugin_mixins/opensearch/api_configs.rb (hunk 4 above)
Comment (VijayanB) [reply lib/logstash/plugin_mixins/opensearch/api_configs.rb:13]: Ack

File: spec/fixtures/_nodes/7x.json (hunk 7 above)
Comment (VijayanB) [reply spec/fixtures/_nodes/7x.json:7]: Yeah. I removed only references in text and tests. The fill name is used somewhere very deep. Will remove this in next PR.

File: lib/logstash/outputs/opensearch.rb (hunk 1 above)
Comment (jmazanec15) [reply lib/logstash/outputs/opensearch.rb]: Sounds good

**Review by jmazanec15 — APPROVED** [review]

**Review by vamshin — APPROVED** [review]
LGTM! Thanks

**Pull request merged**